runtime/libffi/z/ffi64.c

```diff
@@ -861,7 +907,7 @@ ffi_prep_cif_machdep(ffi_cif *cif)
         FFI_ASSERT (0);
         break;
     }
-
+  cif->rtype->type = ffi_check_struct_for_complex(cif->rtype);
```

Comment:
Should this happen in (or before) the `FFI_TYPE_STRUCT` case above? The switch statement above makes decisions based on the return type which may be altered here.

Reply:
No, I think this is right place, the switch statement above sets up the flags on the cif object which will be used to jump to a code in assembly to handle return type. Changing of type for special case should be done after that.

Reply:
It should be after - Previous switch statement checks out the return type to setup flags that helps jump to appropriate code to handle return type in the Assembly code - We need this flag as well the returntype to handles such special return types correctly.

Reply:
I don't understand. If this changes the return type from `FFI_TYPE_STRUCT` to `FFI_TYPE_STRUCT_FF` or `FFI_TYPE_STRUCT_DD`, then how can the behavior for `FFI_TYPE_STRUCT` established by the switch above be appropriate?

Reply:
Flag set there is being checked here : https://github.com/eclipse-openj9/openj9/blob/1be5b9841289086f383e4999e8ec2b930031d193/runtime/libffi/z/sysvz64.s#L827-L834 to jump to return type (Primitive / Struct in GPR / Struct in Memory). When we jump to the handling Struct in memory - it will look into the type to handle two floats or two doubles. The type check happens here : https://github.com/eclipse-openj9/openj9/blob/1be5b9841289086f383e4999e8ec2b930031d193/runtime/libffi/z/sysvz64.s#L865-L871